onnxruntime/test/python/onnxruntime_test_collective.py

```diff
@@ -154,42 +155,50 @@ def _create_alltoall_ut_model_for_boolean_tensor(
-                model.SerializeToString(),
-                providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
-                provider_options=[{"device_id": str(rank)}, {}],
-            )
+    @parameterized.expand(
+        [
+            (np.float32, TensorProto.FLOAT),
+        ]
+    )
+    def test_all_reduce(self, np_elem_type, elem_type):
+        model = self._create_allreduce_ut_model((128, 128), elem_type)
+        rank, size = self._get_rank_size()
+        ort_sess = ort.InferenceSession(
+            model.SerializeToString(),
+            providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
+            provider_options=[{"device_id": str(rank)}, {}],
+        )
 
-            input = np.ones((128, 128), dtype=np_elem_type)
-            outputs = ort_sess.run(None, {"X": input})
-            assert np.allclose(outputs[0], size * input)
-
-    def test_all_gather(self):
-        for np_elem_type, elem_type, communication_elem_type in ((np.float32, TensorProto.FLOAT, TensorProto.FLOAT),):
-            model = self._create_allgather_ut_model((128, 128), 0, elem_type, communication_elem_type)
-            rank, size = self._get_rank_size()
-            ort_sess = ort.InferenceSession(
-                model.SerializeToString(),
-                providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
-                provider_options=[{"device_id": str(rank)}, {}],
-            )
+        input = np.ones((128, 128), dtype=np_elem_type)
+        outputs = ort_sess.run(None, {"X": input})
+        assert np.allclose(outputs[0], size * input)
 
-            input = np.ones((128, 128), dtype=np.float32) * rank
-            outputs = ort_sess.run(None, {"X": input})
+    @parameterized.expand(
+        [
+            (np.float32, TensorProto.FLOAT, TensorProto.FLOAT),
+        ]
+    )
+    def test_all_gather(self, np_elem_type, elem_type, communication_elem_type):
+        model = self._create_allgather_ut_model((128, 128), 0, elem_type, communication_elem_type)
+        rank, size = self._get_rank_size()
+        ort_sess = ort.InferenceSession(
+            model.SerializeToString(),
+            providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
+            provider_options=[{"device_id": str(rank)}, {}],
+        )
 
-            expected_output = np.zeros((128, 128), dtype=np_elem_type)
-            for _ in range(size - 1):
-                expected_output = np.concatenate((expected_output, np.ones((128, 128), dtype=np_elem_type) * (_ + 1)))
-            np.testing.assert_allclose(outputs[0], expected_output, err_msg="all gather on axis0: result mismatch")
+        input = np.ones((128, 128), dtype=np.float32) * rank
+        outputs = ort_sess.run(None, {"X": input})
+
+        expected_output = np.zeros((128, 128), dtype=np_elem_type)
+        for _ in range(size - 1):
+            expected_output = np.concatenate((expected_output, np.ones((128, 128), dtype=np_elem_type) * (_ + 1)))
```

Comment:
I honestly don't understand this computation. Could you give an example as a comment in code? E.g.,
```
# Assume you have 2 devices.
# Device 0 has 2-D tensor x=[ [0, 1], [2, 3] ]
# Device 1 has 2-D tensor y=[ [4, 5], [6, 7] ]
# AllToAll(x, y) may produce
# [ [?, ?], [?, ?] ] on device 0 and
# [ [?, ?], [?, ?] ] on device 1.
```

Reply:
You are commenting in all gather implementation but your comment reads AllToAll. Little confused about where this comment was intended for AllGather or AllToAll?

Reply:
Added comments on inputs/outputs for all operations.